### DISCUSSION TITLE
How can I force Prisma to disconnect in an integration test?

### DISCUSSION THREAD

**Discussion by aeddie-zapidhire** [prisma/prisma#11613] [Q&A] [answered]
I'm trying to test a health-check API (`GET /hc`).

Using an Express router, I'm doing something like the following:

```typescript
    router.get('/hc', (req: Request, res: Response, next: NextFunction) => {
      db.$executeRaw(sql`SELECT NOW()`)
        .then((r) => res.json({ status: 'ok' }))
        .catch((e) => next(e))
    })
```

The problem we are trying to solve is that our running container "somehow" temporarily lost permissions with RDS, so we are adding a simple query to the health check to just check it's ok. But, try to write an integration test is proving to be a challenge.

So far I'm doing a test something akin to:

```typescript
    it.skip('should error if database down', async () => {
      await db.$disconnect()

      try {
        await axios(`${API_HOST}/hc`)
        throw new Error('should throw')
      } catch (e: any) {
        // console.log('e.respsonse', e.respsonse)
        // console.log('e.request', e.request)
        // console.log('e', e)
        equal(e.response.status, 503)
      }
    })
```

But I think what's happening is that Prisma is reconnecting when I try to hit the health check.

Any thoughts on what I can do to kill the Prisma connection after I mount the API?

**Comment by pantharshit00** [comment, marked as answer]
`$disconnect()` returns a Promise. Can you please try waiting for that promise?

**Reply by aeddie-zapidhire** [reply]
@pantharshit00 apologies, that's a mistake in my snippet above. Yes, I am awaiting the promise. I'll correct the snippet.

**Reply by pantharshit00** [reply]
Is Prisma still trying to reconnect after you await the `$disconnect` Promise? Can you please also tell me which database you are using? 

Can you please also make sure you are not calling `$connect` anywhere in your code after you have called disconnect?

**Reply by janpio** [reply]
If you send a query again, Prisma will automatically reconnect. There is no way to prevent that - as that is the job of Prisma. When you use it, you want to talk to a database.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
